lambda-deployment-orca/src/main/java/com/amazon/aws/spinnaker/plugin/lambda/traffic/LambdaTrafficUpdateVerificationTask.java

```diff
@@ -64,7 +68,38 @@ public TaskResult execute(@NotNull StageExecution stage) {
             return formErrorTaskResult(stage,op.getErrors().getMessage());
         }
 
+        if (!"$WEIGHTED".equals(stage.getContext().get("deploymentStrategy"))) {
+            boolean valid = validateWeights(stage);
+            if (!valid) {
+                formErrorTaskResult(stage, "Could not update weights in time");
+                return TaskResult.builder(ExecutionStatus.TERMINAL).outputs(stage.getOutputs()).build();
+            }
+        }
+
         copyContextToOutput(stage);
         return taskComplete(stage);
     }
+
+    private boolean validateWeights(StageExecution stage) {
+        utils.await(30000);
+        AliasRoutingConfiguration weights = null;
+        long startTime = System.currentTimeMillis();
+        LambdaTrafficUpdateInput inp = utils.getInput(stage, LambdaTrafficUpdateInput.class);
+        boolean status = true;
+        do {
+            utils.await(3000);
```

Comment:
Same questions here.

Reply:
This is just the time to wait until sending another request to clouddriver with retrieveLambdaFromCache - any suggestion on how much time should it wait?

Reply:
I bet this needs to be configurable.  Depending where spinnaker is running in vs. the target account, and who knows what other network excitement is happening in between...

Reply:
Changed to use config value of `cacheRefreshRetryWaitTime` which is used for the same thing in other parts of the code